### PR TITLE
Added ignoreList option

### PR DESCRIPTION
**Pull request by IvanKalinin**
Added an option that allows us to ignore certain selectors with increase of specificity

### CODE REVIEW

**Comment by MadLittleMods** [comment]
@IvanKalinin Sorry for the delay 😞 

Any reason not to use `postcss-plugin-context`, example here, https://github.com/MadLittleMods/postcss-increase-specificity#only-apply-to-certain-sections-of-styles

Or you could use [`postcss-raw`](https://github.com/MadLittleMods/postcss-raw) to remove the selector while increasing the specificity.

```js
var postcss = require('postcss');
var raw = require('postcss-raw');
var increaseSpecificity = require('postcss-increase-specificity');

var fs = require('fs');

var mycss = fs.readFileSync('input.css', 'utf8');

// Process your CSS with postcss-reverse-media
var output = postcss([
		raw.inspect(/*options*/),
		increaseSpecificity(/*options*/),
		raw.write(/*options*/)
                // ... your other plugins
	])
	.process(mycss)
	.css;

console.log(output);
```

```css
@raw {
  .selector {
    background: red;
  }
}

.another-selector {
  background: green;
}
```

wdyt?

**Comment by IvanKalinin** [comment]
@MadLittleMods Sorry, now I missed your reply.
According to your README example `postcss-plugin-context` provides ability to increase specificity for certain section. But ignoreList does opposite thing. It exclude some sections (selectors) from processing.
`postcss-raw` requires additional wrappers in existed stylesheets. So I'm not sure that those plugins fit in the need
But `postcss-raw` might be an option

File: test/fixtures/ignore-list.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+.anotherSelector.selector {
+  background: black;
+}
+
+.anotherSelector .selector{
+  background: white;
+}
+
```
Comment (MadLittleMods) [inline test/fixtures/ignore-list.css:36]: Let's change the selector to ignore, `.ignoreselector` so it's easier to understand

We also need to add a test for `.ignoreselector-notme`  (doesn't pass AFAICT)

File: index.js [hunk 2]
```diff
@@ -10,6 +10,17 @@ var CSS_ESCAPED_TAB = '\\9';
 
 function increaseSpecifityOfRule(rule, opts) {
 	rule.selectors = rule.selectors.map(function(selector) {
+		// Ignore a selector if it is in the ignore list
+		if (opts.ignoreList.length > 0) {
+			var isInIgnoreList = opts.ignoreList.some(function(item) {
+				var regEx = new RegExp(item.replace('.', '\\.') + '\\b', 'gi');
```
Comment (MadLittleMods) [inline index.js:16]: We should precompute these regexes outside of the `rule.selectors.map` loop.

---

Some small nits

 - `opts.ignoreList.some(function(ignoreItem) {`
 - `regEx` -> `ingoreRe`

---

We should note in the docs that this doesn't handle complex ignore selectors.

ex.

 - Ignore `.foo.bar`
 - Selector: `.foo.QUX.bar` would still increase specificity even though it is technically under the `.foo.bar` scope

**Review by MadLittleMods — CHANGES_REQUESTED** [review]
We also need to update the readme

**Comment by MadLittleMods** [comment]
@IvanKalinin The alternative solutions do involve more markup but it may also be easier to understand how it works if the ignore context is in the CSS.

Another solution could be to ignore any rule with a comment above `// ignore-postcss-increase-specificity`. Just a thought.

I added some review comments in any case to move this forward.

**Comment by MadLittleMods** [comment]
Closing in favor of inline comment ignore solution, https://github.com/MadLittleMods/postcss-increase-specificity/issues/17

**Pull request closed without merging**